### ISSUE TITLE
Consider moving System.Net.Mime.MedaTypeNames to System.Net.Primitives package

### ISSUE THREAD

**Issue by peteraritchie** [dotnet/corefx#26937]
`System.Net.Mime.MedaTypeNames` is used outside of Mail and is much a part of HTTP communications.  Without a `System.Net.Http.Primitives` package, `System.Net.Primitives` seems like a better place for types like this to exist.  HTTP in .NET Standard supports HTTP in versions lower than 2.0; but with `System.Net.Mime.MediaTypeNames` in its current location, it's not available for .NET Standard code lower than 2.0.    Moving it to Primitives would make it available lower than 2.0 as well as 2.0.

**Comment by karelz** [comment]
Moving the type into another assembly won't make it magically available on lower .NET Standard versions. Moving types around is possible, but we would need a very good reason as it leads to more type forwarders, etc.

Patching existing .NET Standards is actually non-trivial work, and likely requires also tooling updates, etc. - overall it is not worth the effort unless it is something super-critical (I don't think we did it yet at all).

Closing as won't fix.

**Comment by peteraritchie** [comment]
I suggested the Primitives not because moving a type magically makes anything work, Primitives targets lower versions of .NET Standard.  I also chose primitives because `MediaTypeNames` is just a class with nested classes with only const string fields.  Compared to the other types in Primitives, a much better fit.

I wasn't expecting it to be non-trivial; I did think its the right thing to do regardless of how much work it is.  Just have a look at how many "application/json" literals are floating around in the frameworks.  The amount of work to make some something better or better align with principles behind something shouldn't be the deciding factor--otherwise we'd never make any changes, because it's all non-trivial work.

**Comment by karelz** [comment]
> Primitives targets lower versions of .NET Standard.

In general, we stopped shipping NuGet packages targeting lower versions which also ship as part of the platform - it creates giant mess (incl. complexity of what can use newer APIs in other packages vs. not) and test-matrix growth exponentially.
Technically speaking, I don't think that by moving the type, it would automatically light up for the version targeting lower .NET Standard - it would just show up in the latest .NET Core / future .NET Standard.

> I did think its the right thing to do regardless of how much work it is

Right thing is often a very relative term. Cost is important part of it as well - anything you pay is opportunity cost. You have to push out something else. Overall it is better to focus on things with higher impact - i.e. prioritize.

> Just have a look at how many "application/json" literals are floating around in the frameworks. The amount of work to make some something better or better align with principles behind something shouldn't be the deciding factor--otherwise we'd never make any changes, because it's all non-trivial work.

I also look at the amount of other pain points, bugs and things that prevent customers to do things and prioritize among them. The value of adding "application/json" is convenience, there is no measurable perf improvement, or unblocked scenario (workaround is trivial). Such features and additions are great for vNext (assuming low cost), but the cost of porting them to lower versions (incl. long-term maintenance) is not worth it based on our experience.
I would also argue that almost every change in CoreFX repo makes something better and leads to better alignment with principles, not just changes like this one. For example compatibility between platforms (Windows vs. Linux, or .NET Core vs. .NET Framework vs. Mono vs. UWP)
